design/one-pager-change-logs.md

```diff
@@ -0,0 +1,310 @@
+# Change Logs
+
+* Owner: Jared Watts (@jbw976)
+* Status: Proposed
+* Issue: [#5477](https://github.com/crossplane/crossplane/issues/5477)
+
+## TL;DR
+
+* To build further trust and confidence in Crossplane, we will log every
+  operation that is performed on all managed resources
+* The managed reconciler will generate a record of each change made to an
+  external system, as well as the state of the resource before and after the
+  operation
+* Each change log entry will be sent over gRPC to a sidecar container in the
+  provider pod, where it will be persisted as a standard pod log
+* Pod logs are a very standard location that allows many different tools to view
+  and interact with the change log data
+* Change log data is the only content in these logs, so it can also easily be
+  sent to observability systems for further processing and analysis using a
+  component like the OpenTelemetry Collector
+
+## Background
+
+Crossplane is deployed to manage the most critical infrastructure resources for
+many of the organizations that have chosen to adopt it. Crossplane acts in a
+continuously reconciling loop to drive the state of external systems and
+resources to match the desired state declared by the users platform. If changes
+between the observed state of the real world and the desired state from the
+users are detected, Crossplane will automatically update the resources to remove
+this difference.
+
+This means that it is possible for Crossplane to make changes to external
+systems even when no user is actively interacting with the control plane.
+Consider the scenario when someone updates a resource outside of Crossplane,
+e.g. via the AWS console or `gcloud` CLI. When Crossplane detects this
+"configuration drift", it will enforce its "source of truth" to immediately
+correct this unexpected change without any user interaction with the control
+plane.  Currently, there is no mechanism for users of Crossplane to have
+visibility into these types of "unexpected" changes that Crossplane is making on
+their behalf.
+
+With Crossplane acting continuously and autonomously to update critical
+infrastructure, it is vital for users to have insight into the operations being
+performed, so they can build and maintain a strong sense of confidence and trust
+in their control planes.
```

Comment:
This thought comes to mind, playing devil's advocate: if this feature is groundbreaking and critical, why not just look at cloud provider audit logs? They exist. Why are they not enough?

Reply:
> Why are they not enough?

Compared to cloud change logs:

* Not all systems Crossplane has providers for will have audit logs.
* Cloud change logs are homogenous, so it's easier to configure them all to go to some single pane of glass (compared to configuring Google audit logs, AWS audit logs, etc).
* Cloud change logs take away any guesswork as to what changes Crossplane made, as opposed to trying to use cloud identity, or HTTP user agents etc to identify Crossplane changes.

You can probably get pretty far just setting up cloud audit logs, but I don't think that negates the value of this feature.

Reply:
All good arguments. Why not write it down? :)

Reply:
Added a new alternatives considered section for this cloud providers audit logs idea too 🤓